packages/core/src/components/graphs/scatter.ts

```diff
@@ -54,19 +54,19 @@ export class Scatter extends Component {
 		dotsEnter.merge(dots)
 			.raise()
 			.classed("dot", true)
-			.classed("filled", filled)
-			.classed("unfilled", !filled)
+			.classed("filled", d => this.model.getIsFilled(d.datasetLabel, d.label, d.value, d, filled))
+			.classed("unfilled", d => !this.model.getIsFilled(d.datasetLabel, d.label, d.value, d, filled))
```

Comment:
If we're calling a function here we're better off storing the value and not calling that function twice

Reply:
this is true, but I think I need the specific datapoint from the d3 callback